File: src/models/cointoss.rs
use serde::{Deserialize, Serialize};

#[derive(Default, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
pub struct Cointoss {
    tossed: bool,
	away_picked_heads: bool,
	result_heads: bool,
	home_receives: bool
	
}

impl Cointoss {
	pub fn new() -> Cointoss {
        Default::default()
    }    
}

File: java/src/main/java/com/foul/gameserver/domain/decission/random/CoinToss.java
package com.foul.gameserver.domain.decission.random;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@Getter
@EqualsAndHashCode
@RequiredArgsConstructor
@ToString
public final class CoinToss {
	
	Logger logger = LoggerFactory.getLogger(CoinToss.class);

	boolean tossed;
	boolean awayPickedHeads;
	boolean resultHeads;
	boolean homeReceives;
  
}